File: .lefthook/post-checkout/cleanup.js
const fs = require("fs");
const path = require("path");

if (process.argv[3] === "1") {
  console.log("Cleaning up build artifacts and dependencies...");

  const targetDir = path.join(process.cwd(), "target");
  if (fs.existsSync(targetDir)) {
    console.log("Removing Rust target directory...");
    fs.rmSync(targetDir, { recursive: true, force: true });
  }

  const nodeModulesDir = path.join(process.cwd(), "frontend", "node_modules");
  if (fs.existsSync(nodeModulesDir)) {
    console.log("Removing pnpm node_modules directory...");
    fs.rmSync(nodeModulesDir, { recursive: true, force: true });
  }
  console.log("Cleanup completed.");
} else {
  console.log("No branch switch detected. Skipping cleanup.");
}

File: .lefthook/post-checkout/cleanup.rs
use std::env;
use std::fs;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args[3] == "1" {
        println!("Cleaning up after checkout...");
        let current_dir = env::current_dir().unwrap();
        let targetDir = Path::new(&current_dir).join("target");
        if targetDir.exists() {
            println!("Removing target directory...");
            fs::remove_dir_all(&targetDir).expect("Failed to remove target directory");
        } else {
            println!("Target directory does not exist, nothing to clean up.");
        }
        let node_modules_dir = Path::new(&current_dir)
            .join("frontend")
            .join("node_modules");
        if node_modules_dir.exists() {
            println!("Removing node_modules directory...");
            fs::remove_dir_all(&node_modules_dir).expect("Failed to remove node_modules directory");
        } else {
            println!("node_modules directory does not exist, nothing to clean up.");
        }
    } else {
        println!("No cleanup needed.");
    }
}
